bazel: incorrect geospatial results in logictest

I hit this failure when using bazel. This is with no local changes (5ad21e3896).
```
        --- FAIL: TestLogic/local/geospatial_zm (0.23s)
            logic.go:2438: 
                 
                testdata/logic_test/geospatial_zm:381: SELECT st_length('LINESTRING M(0 0 -25, 1 1 -50, 2 2 0)')
                expected:
                    2.82842712474619
                    
                but found (query options: "") :
                    82.2650791318405
```
Test passes with make.

--- a/pkg/geo/geos/geos.go
+++ b/pkg/geo/geos/geos.go
@@ -130,6 +130,12 @@ func findLibraryDirectories(flagLibraryDirectoryValue string, crdbBinaryLoc stri
 	if flagLibraryDirectoryValue != "" {
 		locs = append(locs, flagLibraryDirectoryValue)
 	}
+	// Account for the libraries to be in a bazel runfile path.
+	if bazel.BuiltWithBazel() {
+		if p, err := bazel.Runfile(path.Join("c-deps", "libgeos", "lib")); err == nil {
+			locs = append(locs, p)
+		}
+	}
 	locs = append(
 		append(
 			locs,
@@ -137,12 +143,6 @@ func findLibraryDirectories(flagLibraryDirectoryValue string, crdbBinaryLoc stri
 		),
 		findLibraryDirectoriesInParentingDirectories(cwd)...,
 	)
-	// Account for the libraries to be in a bazel runfile path.
-	if bazel.BuiltWithBazel() {
-		if p, err := bazel.Runfile(path.Join("c-deps", "libgeos", "lib")); err == nil {
-			locs = append(locs, p)
-		}
-	}
 	return locs
 }
 

